File: admin/src/main/java/top/zzf4/blog/entity/model/AccessLog.java
package top.zzf4.blog.entity.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("access_logs")
public class AccessLog {
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("ip_address")
    private String ipAddress;

    @TableField(value = "access_time", fill = FieldFill.INSERT)
    private LocalDateTime accessTime;

    @TableField("user_agent")
    private String userAgent;

    @TableField("request_url")
    private String requestUrl;

    @TableField(exist = false)
    private String city;
}

File: blog-backend/internal/model/access_log.go
package model

// AccessLog 对应于数据库中的 `access_logs` 表
type AccessLog struct {
	BaseModel  `gorm:"embedded"`
	IPAddress  string `gorm:"column:ip_address;not null" json:"ipAddress"` // IP 地址
	UserAgent  string `gorm:"column:user_agent" json:"userAgent"`          // 用户代理
	RequestURL string `gorm:"column:request_url" json:"requestUrl"`        // 请求 URL
	City       string `gorm:"-" json:"city,omitempty"`                     // 城市 (不映射到数据库，仅用于数据传输)
}

// TableName 指定 GORM 使用的表名
func (AccessLog) TableName() string {
	return "access_logs"
}
